### PR TITLE
Fix parameterization issues

### PR DESCRIPTION
**Pull request by saqadri**
Fix parameterization issues

Now that we have non-text input model parsers, we can see some issues in our parameterization.

1) ASR and Image-to-Text modelparsers should NOT be `ParameterizedModelParser` instances, since their inputs cannot be parameterized.

2) Parameter resolution logic shouldn't throw errors in the case where it's parsing prompts belonging to regular model parsers.

Test Plan:
Ran the translation prompt for this aiconfig:

```
prompts: [
{
      "name": "image_caption",
      "input": {
        "attachments": [
          {
            "data": "https://s3.amazonaws.com/lastmileai.aiconfig.public/uploads/2024_1_10_18_41_31/1742/bear_with_honey.png",
            "mime_type": "image/png"
          },
          {
            "data": "https://s3.amazonaws.com/lastmileai.aiconfig.public/uploads/2024_1_10_18_41_31/7275/fox_in_forest.png",
            "mime_type": "image/png"
          }
        ]
      },
      "metadata": {
        "model": {
          "name": "HuggingFaceImage2TextTransformer",
          "settings": {
            "max_new_tokens": 10,
            "model": "Salesforce/blip-image-captioning-base"
          }
        },
        "parameters": {}
      },
      "outputs": [
        {
          "output_type": "execute_result",
          "execution_count": 0,
          "data": "a bear sitting on a rock eating honey",
          "metadata": {}
        },
        {
          "output_type": "execute_result",
          "execution_count": 1,
          "data": "a red fox in the woods",
          "metadata": {}
        }
      ]
    },
    {
      "name": "translation",
      "input": "Once upon a time, in a lush and vibrant forest, there lived a magnificent creature known as the Quick Brown Fox. This fox was unlike any other, possessing incredible speed and agility that awed all the animals in the forest. With its fur as golden as the sun and its eyes as sharp as emeralds, the Quick Brown Fox was admired by everyone, from the tiniest hummingbird to the mightiest bear. The fox had a kind heart and would often lend a helping paw to those in need. The Quick Brown Fox had a particular fondness for games and challenges. It loved to test its skills against others, always seeking new adventures to satisfy its boundless curiosity. Its favorite game was called \"The Great Word Hunt,\" where it would embark on a quest to find hidden words scattered across the forest. \n\nOne day it got very old and died",
      "metadata": {
        "model": {
          "name": "HuggingFaceTextTranslationTransformer",
          "settings": {
            "model": "Helsinki-NLP/opus-mt-en-fr",
            "max_length": 100,
            "min_length": 50,
            "num_beams": 1
          }
        },
        "parameters": {}
      }
    }
]
```

Before:

```
File "/opt/homebrew/lib/python3.11/site-packages/aiconfig/util/params.py", line 235, in get_prompt_template
    raise Exception(f"Cannot get prompt template string from prompt: {prompt.input}")
Exception: Cannot get prompt template string from prompt: attachments=[Attachment(data='https://s3.amazonaws.com/lastmileai.aiconfig.public/uploads/2024_1_10_18_41_31/1742/bear_with_honey.png', mime_type='image/png', metadata=None), Attachment(data='https://s3.amazonaws.com/lastmileai.aiconfig.public/uploads/2024_1_10_18_41_31/7275/fox_in_forest.png', mime_type='image/png', metadata=None)] data=None
```

After:
* Beautiful translation

---
Stack created with [Sapling](https://sapling-scm.com). Best reviewed with [ReviewStack](https://reviewstack.dev/lastmile-ai/aiconfig/pull/866).
* #826
* __->__ #866
* #863

### CODE REVIEW

File: python/src/aiconfig/util/params.py [hunk 1]
```diff
@@ -224,13 +224,12 @@ def get_prompt_template(prompt: Prompt, aiconfig: "AIConfigRuntime"):
...
         return model_parser.get_prompt_template(prompt, aiconfig)
-
-    if isinstance(prompt.input, str):
+    elif isinstance(prompt.input, str):
         return prompt.input
     elif isinstance(prompt.input.data, str):
         return prompt.input.data
     else:
-        raise Exception(f"Cannot get prompt template string from prompt: {prompt.input}")
+        return None
```
Comment (rholinshead) [inline python/src/aiconfig/util/params.py:232]: Let's update the return type from str to union str and None

**Review by rholinshead — APPROVED** [review]
Accepting to unblock but:
- please update the return type for get_prompt_template
- can you add an issue to update typescript side for consistency? We should get in the habit of this whenever sdk changes are made

File: python/src/aiconfig/util/params.py [hunk 2]
```diff
@@ -243,6 +242,9 @@ def collect_prompt_references(current_prompt: Prompt, ai_config: "AIConfigRuntim
             break
 
         prompt_input = get_prompt_template(previous_prompt, ai_config)
+        if prompt_input is None:
```
Comment (Ankush-lastmile) [inline python/src/aiconfig/util/params.py:245]: nit: comment saying something like "not all model parsers are parameterizable" would be nice, and where ryan commented

**Comment by saqadri** [comment]
Filed this issue for TS SDK: https://github.com/lastmile-ai/aiconfig/issues/868

The comments are non-blocking so I will resolve on top of stack. I need to land this to publish a package update

**Pull request merged**